backend/cellguide/pipeline/canonical_marker_genes/canonical_markers.py

```diff
@@ -0,0 +1,285 @@
+            if symbol and symbol in self.wmg_human_genes:
+                name = gene.rdfs_label or self.gene_id_to_name.get(symbol, symbol)
+                gene_symbols.append(symbol)
+                gene_names.append(name)
+        return gene_symbols, gene_names
+
+    def _get_references(
+        self, references: list[Reference], doi_to_citation: Dict[str, str]
+    ) -> Tuple[list[str], list[str]]:
+        """
+        Extracts the DOIs and citations from the given list of references.
+        This function cleans the DOI and gets the title and formatted citation from the DOI.
+
+        Arguments
+        ---------
+        references - list[Reference]
+            a list of references for a particular ASCTB table entry.
+        doi_to_citation - Dict[str,str]
+            a dictionary mapping DOIs to citations.
+
+        Returns
+        -------
+        refs - list[str]
+            The list of cleaned DOIs extracted from the input references.
+        titles - list[str]
+            The list of titles extracted from the input references.
+        """
+
+        refs = []
+        titles = []
+        for ref in references:
+            doi = clean_doi(ref.doi)
+            if doi:
+                if doi not in doi_to_citation:
+                    title = get_title_and_citation_from_doi(doi)
+                    doi_to_citation[doi] = title
+                else:
+                    title = doi_to_citation[doi]
+                refs.append(doi)
+                titles.append(title)
+        return ";;".join(refs), ";;".join(titles)
+
+    def get_processed_asctb_table_entries(self) -> Dict[str, ParsedAsctbTableEntry]:
+        """
+        Processes the ASCTB table entries and returns a dictionary mapping cell type ontology term IDs to
+        ParsedAsctbTableEntry objects. The processing involves extracting relevant information from the ASCTB data
+        such as tissue ID, gene symbols and names, DOIs and citations, and cell type ontology term IDs. It also
+        involves cleaning and formatting the extracted data.
+
+        Returns
+        -------
+        Dict[str, ParsedAsctbTableEntry]
+            A dictionary mapping cell type ontology term IDs to ParsedAsctbTableEntry objects.
+        """
+
+        # DOI to citation mapping
+        doi_to_citation = {}
+
+        hashed_entries_seen = []
+        parsed_table_entries = []
+
+        for i, tissue in enumerate(self.asctb_data):
+            version = self.asctb_data[tissue]["metadata"]["version"]
+            logger.info(
+                f"Getting processed ASCTB table entry for {tissue} tissue ({version}) ({i+1}/{len(self.asctb_data)})"
+            )
+            data = self.asctb_data[tissue]["data"]
+
+            for row in data:
+                cell_types = [i["id"] for i in row["cell_types"] if i["id"].startswith("CL:")]
+                if not cell_types or not row["biomarkers_gene"]:
+                    continue
```

Comment:
i don't think it's necessary - there are a ton of ASCTB rows that only have anatomical structures or no biomarkers so it would just add noise to the logs